src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt

```diff
@@ -61,47 +61,57 @@ class RestrictedPatientsService(
   fun migrateInPatient(migrateIn: MigrateInRequest): RestrictedPatientDto {
     checkNotExistingPatient(migrateIn.offenderNo)
 
-    val latestMovements = prisonApiGateway.getLatestMovements(migrateIn.offenderNo)
-    val dischargeData = getExistingRestrictedPatientDischargeData(latestMovements, migrateIn.offenderNo)
+    val dischargeData = getLatestMovement(migrateIn.offenderNo)
+      .also { checkLatestMovementOkForDischarge(it, migrateIn.offenderNo) }
+      .let { getExistingRestrictedPatientDischargeData(it, migrateIn.offenderNo) }
```

Comment:
I thought I needed this refactor to re-use some validation - turned out not but forgot to take it out.